conans/client/graph/graph_binaries.py

```diff
@@ -274,7 +274,9 @@ def package_id_transitive_reqs(node):
         node.id_indirect_prefs = set()  # of PackageReference, avoid duplicates
         neighbors = [d.dst for d in node.dependencies if not d.build_require]
         for neighbor in neighbors:
-            node.id_direct_prefs.add(neighbor.pref)
+            GraphBinariesAnalyzer.package_id_transitive_reqs(neighbor)
+            pref = PackageReference(neighbor.ref, neighbor.pref.id, neighbor.prev)
```

Comment:
I'm not sure if it is a bug or not, but the attribute `neighbor.pref` doesn't contain the update `prev`. Need to double-check if it is intended or not.